cmd/clusterctl/client/config.go

```diff
@@ -302,16 +302,30 @@ func (c *clusterctlClient) templateOptionsToVariables(options GetClusterTemplate
 	}
 
 	// the ControlPlaneMachineCount, if valid, can be used in templates using the ${ CONTROL_PLANE_MACHINE_COUNT } variable.
-	if options.ControlPlaneMachineCount < 1 {
-		return errors.Errorf("invalid ControlPlaneMachineCount. Please use a number greater or equal than 1")
+	if options.ControlPlaneMachineCount == nil {
```

Comment:
I'm wondering if we can validate ControlPlaneMachineCount also when the values comes from env variables.
What about something like 
```

//Check if set through env variable and default to 1 otherwise
if options.ControlPlaneMachineCount == nil {
	if value, err := c.configClient.Variables().Get("CONTROL_PLANE_MACHINE_COUNT"); err != nil {
		options.ControlPlaneMachineCount = utilpointer.Int32Ptr(1)
	} else {
		options.ControlPlaneMachineCount = utilpointer.Int32Ptr(value)
	}
} 

if *options.ControlPlaneMachineCount < 1 {
	return errors.Errorf("invalid ControlPlaneMachineCount. Please use a number greater or equal than 1")
}
c.configClient.Variables().Set("CONTROL_PLANE_MACHINE_COUNT", strconv.Itoa(*options.ControlPlaneMachineCount))
```

Reply:
sounds good. will update